src/reseeding.rs

```diff
@@ -51,43 +55,85 @@ impl<R: Rng, Rsdr: Reseeder<R>> ReseedingRng<R, Rsdr> {
 
     /// Reseed the internal RNG if the number of bytes that have been
     /// generated exceed the threshold.
-    pub fn reseed_if_necessary(&mut self) {
+    pub fn reseed_if_necessary(&mut self) -> Result<(), Error> {
         if self.bytes_generated >= self.generation_threshold {
-            self.reseeder.reseed(&mut self.rng);
+            let _ = self.reseeder.reseed(&mut self.rng).map_err(map_err)?;
+            // Reset number of generated bytes, even if reseeding failes.
+            // This means we don't try to reseed again until a new
+            // `generation_threshold` amount of output is consumed.
             self.bytes_generated = 0;
-        }
+        };
+        Ok(())
+    }
+}
+
+/// An error that can occur when reseeding failed.
+#[cfg(feature="std")]
+#[derive(Debug)]
+pub struct ReseedingError {
```

Comment:
I'm not convinced we need another error type (see #10).

Reply:
An alternative is to add a `ReseedingFailed` variant to the error kinds in `rand_core`.

It is just that I first explored this route (should have made a better PR message that said so).